src/promptflow-tools/promptflow/tools/common.py

```diff
@@ -384,16 +384,19 @@ def normalize_connection_config(connection):
     """
     if isinstance(connection, AzureOpenAIConnection):
         use_key_auth = True
-        try:
-            from promptflow._sdk._constants import ConnectionAuthMode
-            if connection.auth_mode == ConnectionAuthMode.MEID_TOKEN:
-                use_key_auth = False
-        except ImportError as e:
-            if "cannot import name 'ConnectionAuthMode' from 'promptflow._sdk._constants'" in str(e):
-                print("Failed to import ConnectionAuthMode, use key auth by default.")
-                pass
-            else:
-                raise e
+
+        if not connection.api_key:
+            try:
+                from promptflow._sdk._constants import ConnectionAuthMode
+                if connection.auth_mode == ConnectionAuthMode.MEID_TOKEN:
+                    use_key_auth = False
+            except ImportError as e:
+                if "cannot import name 'ConnectionAuthMode' from 'promptflow._sdk._constants'" in str(e):
+                    print("Failed to import ConnectionAuthMode, use meid_token auth by default.")
```

Comment:
Why failed to import `ConnectionAuthMode`, fallback to use auth?

And looks no matter line 390/391 works, will always use auth mode?

Reply:
when api_key is None, I think we should try to use token auth.
currently yes, as we only have key_based and token_based auth_mode.